frontend/ui/trapper.lua

```diff
@@ -167,112 +179,155 @@ function Trapper:UIFeedback(text, show_confirmbox, ok_text, cancel_text)
         -- continue processing
     end
 
-    if text == nil then -- Trapper:UIFeedback()
-        -- let current widget be, just return the fact it has not been dismissed
-        return true
-    end
-
     -- TODO We should try to flush any pending tap, so past
     -- events won't be considered action on the yet to be displayed
     -- widget
 
-    if text == false then -- Trapper:UIFeedback(false)
-        -- End the whole interaction
-        UIManager:close(self.current_widget)
-        UIManager:forceRePaint()
-        -- Reset some properties
-        self.current_widget = nil
-        self.current_widget_is_infomessage = nil
-        self.paused_text = nil
-        self.paused_continue_text = nil
-        self.paused_abort_text = nil
-        return true
-    end
-
     -- We're going to display a new widget, close previous one
     if self.current_widget then
         UIManager:close(self.current_widget)
         -- no repaint here, we'll do that below when a new one is shown
     end
 
-    if show_confirmbox then
-        -- ConfirmBox: both callbacks will resume coroutine
-        self.current_widget = ConfirmBox:new{
-            text = text,
-            ok_text = ok_text,
-            cancel_text = cancel_text,
-            ok_callback = function()
-                coroutine.resume(_coroutine, true)
-            end,
-            cancel_callback = function()
-                coroutine.resume(_coroutine, false)
-            end,
-        }
-        self.current_widget_is_infomessage = false
-    else
-        -- InfoMessage: dismiss_callback will be checked for at start of
-        -- next call
-        self.current_widget = InfoMessage:new{
-            text = text,
-            dismiss_callback = function()
-                coroutine.resume(_coroutine, false)
-            end,
-        }
-        self.current_widget_is_infomessage = true
-    end
-    logger.dbg("Showing", show_confirmbox and "ConfirmBox" or "InfoMessage", text)
+    -- dismiss_callback will be checked for at start of next call
+    self.current_widget = InfoMessage:new{
+        text = text,
+        dismiss_callback = function()
+            coroutine.resume(_coroutine, false)
+        end,
+        is_infomessage = true -- flag on our InfoMessages
+    }
+    logger.dbg("Showing InfoMessage:", text)
     UIManager:show(self.current_widget)
     UIManager:forceRePaint()
+    return true
+end
+
+--[[--
+Override Paused text and button texts on the abort ConfirmBox displayed
```

Comment:
Overrides Paused text and button texts on the abort ConfirmBox

(displayed is redundant and the top function description is aimed at brevity)